src/main/java/net/dries007/tfc/api/events/ProspectEvent.java

```diff
@@ -0,0 +1,56 @@
+package net.dries007.tfc.api.events;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * Event fired when the Prospector's Pickaxe is used.
+ * Carries all the data relative to the result displayed to the player.
+ * One of the two subclasses will be used according to the logical side.
```

Comment:
This is actually standard stuff. I didn't come up with this by myself.
This way you can listen to events only on the side that matters...

However I do agree there should be a `side` attribute there, in case you want to listen to the `ProspectEvent` superclass. In my particular case, I don't, and that's the cleaner way to do it (avoids having to always check if side is a specific value when you're only interested in one branch; having 2 event subclasses introduces no overhead).